ddl/cache.lua

```diff
@@ -0,0 +1,108 @@
+local fiber = require('fiber')
+
+local cache = nil
+
+local CACHE_LOCK_TIMEOUT  = 3
+local SPACE_NAME_IDX      = 1
+local SHARD_FUNC_NAME_IDX = 2
+local SHARD_FUNC_BODY_IDX = 3
+
+-- Build cache and setup "on_replace" trigger.
+--
+-- Cache structure format:
+-- cache = {
+--   spaces = {
+--     'space_name' = {
+--       raw = {},      -- raw sharding metadata, used for ddl.get()
+--       processed = {} -- sharding function that is ready to call or a string with error
```

Comment:
It is a `table` with parsed dot notation (like `{'foo', 'bar'}`) or a `function` ready to call (or a `string` with an error). It is not obvious that 'sharding function' words assume the former two (table or function).

Reply:
fixed